### PR TITLE
Native YouTube video removal.

### PR DESCRIPTION
**Pull request by ryanbourneuk**
This PR:

- Changes how YouTube videos are handled within Storm apps! Instead of them being handled natively, the YouTube link will be opened up within a web view or within Safari (based on the type of link as defined in the bundle per video view).
- Resolves an issue on iPad where web views are not opened correctly.
- Resolves an issue within QuizAreaSelectionViewController where the aspect ratio is not correctly calculated & the image is cut off.
- Makes some properties open / public for unit testing.

### CODE REVIEW

File: ThunderCloud.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -156,12 +157,9 @@
 		B11EAF60219AD27700891792 /* ThunderCloudTests.h in Headers */ = {isa = PBXBuildFile; fileRef = B11EAF5F219AD27700891792 /* ThunderCloudTests.h */; settings = {ATTRIBUTES = (Public, ); }; };
 		B11EAF61219AD2EB00891792 /* ListPage.swift in Sources */ = {isa = PBXBuildFile; fileRef = B13DE1211F0E2A3100C6B9B7 /* ListPage.swift */; };
 		B11EAF62219AD2FD00891792 /* Video.swift in Sources */ = {isa = PBXBuildFile; fileRef = B1B046901F0E352A00FD2FA7 /* Video.swift */; };
-		B11EAF65219AD32400891792 /* YouTubeController.swift in Sources */ = {isa = PBXBuildFile; fileRef = B18E5BBC1F8E37C800037D68 /* YouTubeController.swift */; };
-		B11EAF66219AD32A00891792 /* MultiVideoPlayerViewController.swift in Sources */ = {isa = PBXBuildFile; fileRef = B1B046921F0E388800FD2FA7 /* MultiVideoPlayerViewController.swift */; };
```
Comment (simonmitchell) [inline ThunderCloud.xcodeproj/project.pbxproj]: Can we remove all these files too if they're no longer referenced? :)

File: ThunderCloud/QuizAreaSelectionViewController.swift [hunk 2]
```diff
@@ -31,15 +31,23 @@ class QuizAreaSelectionViewController: UIViewController {
...
-		let imageAspect = question.selectionImage.size.height / question.selectionImage.size.width
-		heightConstraint.constant = imageAspect * imageView.frame.width
-		
+        
 		let imageAnalyser = ImageColorAnalyzer(image: question.selectionImage)
 		imageAnalyser.analyze()
 		circleColor = imageAnalyser.detailColor ?? .black
 	}
+    
+    override func viewDidAppear(_ animated: Bool) {
```
Comment (simonmitchell) [inline ThunderCloud/QuizAreaSelectionViewController.swift]: Does this work in `viewWillAppear`? The user may see the frame changing in `viewDidAppear` :D

File: ThunderCloud/VideoListItem.swift [hunk 3]
```diff
@@ -38,6 +38,16 @@ open class VideoListItem: VideoListItemView {
 	override open func handleSelection(of row: Row, at indexPath: IndexPath, in tableView: UITableView) {
 		
 		guard let videoRow = row as? VideoListItem, let videos = videoRow.videos else { return }
-		videoRow.parentNavigationController?.push(videos: videos)
+        
+        guard let video = videos.first(where: { $0.localeString == StormLanguageController.shared.currentLanguage }) ?? videos.first, let link = video.link else {
```
Comment (simonmitchell) [inline ThunderCloud/VideoListItem.swift]: Guessing this has been okayed with the clients we've chatted this over with too? :D

File: ThunderCloud/QuizAreaSelectionViewController.swift (hunk 2 above)
Comment (ryanbourneuk) [reply ThunderCloud/QuizAreaSelectionViewController.swift]: This change was made in https://github.com/3sidedcube/ThunderCloud/pull/148, as the aspect ratio was being calculated incorrectly while the code was within `viewWillAppear` :)

File: ThunderCloud/VideoListItem.swift (hunk 3 above)
Comment (ryanbourneuk) [reply ThunderCloud/VideoListItem.swift]: That's correct yeah, it's been okayed & is already in production for some apps :D

File: ThunderCloud.xcodeproj/project.pbxproj (hunk 1 above)
Comment (ryanbourneuk) [reply ThunderCloud.xcodeproj/project.pbxproj]: Resolved in 5c46f88f6c90f1a7e4856f92ccb1a4ca3a636c5c, YouTubeController should already be removed :D

**Pull request merged**